docs/requirements/channel-upgradability-requirements.md

```diff
@@ -0,0 +1,224 @@
+6. Execution of `MsgChannelUpgradeTry` succeeds on chain B.
+7. Relayer detects execution of `MsgChannelUpgradeTry` and submits `MsgChannelUpgradeAck` on chain A.
+8. Execution of `MsgChannelUpgradeAck` succeeds on chain A.
+9. Chain A disallows port ID `transfer`, channel ID `channel-0` to be upgraded (now that upgrade has completed, then we can disallow future upgrades until approved again by governance or group proposal).
+10. Relayer detects execution of `MsgChannelUpgradeAck` and submits `MsgChannelUpgradeConfirm` on chain B.
+11. Execution of `MsgChannelUpgradeConfirm` succeeds on chain B.
+12. Chain B disallows port ID `transfer`, channel ID `channel-0` to be upgraded (now that upgrade has completed, then we can disallow future upgrades until approved again by governance or group proposal).
+
+Exception flows:
+
+- Proposal passes on chain B after a relayer has submitted `MsgChannelUpgradeInit` on chain A. This is a crossing hello scenario, since chain B has also automatically executed `MsgChannelUpgradeInit`. Relayer can submit `MsgChannelUpgradeTry` on chain B to continue the handshake.
+- Proposal passes on chain B after the initiation upgrade on chain A has passed, then when a relayer attempts to submit `MsgChannelUpgradeTry` on chain A, then the message will fail, the upgrade will be aborted and the state of the channel on chain B will be reverted to `OPEN`.
+- Proposal on chain A passes after the counterparty upgrade timeout specified on chain B, then upgrade will not go through and state on chain B will be rolled back to `OPEN`. If the upgrade initiation timeout has not passed yet on chain B, then a relayer can submit a new `MsgChannelUpgradeInit` on chain A with a new counterparty upgrade timeout.
+
+### Permissioned upgrade on chain A and B
+
+Normal flow:
+
+1. Proposal is submitted on chain A to allow upgrade and initiate upgrade for port ID `transfer`, channel ID `channel-0`.
+2. Proposal is submitted on chain B to allow upgrade and initiate upgrade for port ID `transfer`, channel ID `channel-0`.
+3. Proposal passes on chain A and:
+    a. it allows port ID `transfer`, channel ID `channel-0` to be upgraded;
+    b. it executes `MsgChannelUpgradeInit` to upgrade channel version from `ics20-1` to `{"fee_version":"ics29-1","app_version":"ics20-1"}`.
+4. Proposal passes on chain B and:
+    a. it allows port ID `transfer`, channel ID `channel-0` to be upgraded;
+    b. it executes `MsgChannelUpgradeInit` to upgrade channel version from `ics20-1` to `{"fee_version":"ics29-1","app_version":"ics20-1"}`.
+5. Relayer detects execution of `MsgChannelUpgradeInit` on chain A and submits `MsgChannelUpgradeTry` on chain B proposing to upgrade channel version to `{"fee_version":"ics29-1","app_version":"ics20-1"}`. Since chain B has already executed `MsgChannelUpgradeInit`, the we are in a crossing hello scenario.
```

Comment:
```suggestion
5. Relayer detects execution of `MsgChannelUpgradeInit` on chain A and submits `MsgChannelUpgradeTry` on chain B proposing to upgrade channel version to `{"fee_version":"ics29-1","app_version":"ics20-1"}`. Since chain B has already executed `MsgChannelUpgradeInit`, we are in a crossing hello scenario.
```